### PR TITLE
layers: Use emplace_back where possible

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Basic attempt to use `emplace_back` where it should have an obvious benefits

prepared to criticism and "well technically in c++... " responses :shield:

### CODE REVIEW

**Review by arno-lunarg — APPROVED** [review]
image

File: layers/error_message/logging.cpp [hunk 1]
```diff
@@ -377,7 +377,7 @@ void DebugReport::BeginCmdDebugUtilsLabel(VkCommandBuffer command_buffer, const
     if (nullptr != label_info && nullptr != label_info->pLabelName) {
         auto *label_state = GetLoggingLabelState(&debug_utils_cmd_buffer_labels, command_buffer, /* insert */ true);
         assert(label_state);
-        label_state->labels.push_back(LoggingLabel(label_info));
+        label_state->labels.emplace_back(label_info);
```
Comment (artem-lunarg) [inline layers/error_message/logging.cpp:380]: IMO this gets less readable. Before it was clear we put LoggingLabel objects, now the first impression we put VkDebugUtilsLabelEXT pointers there.

One of the reasons of introduction of emplace methods is to facilitate meta programming where specific types are not known or even not visible and require decltype.  In that case emplace does not reduce code clarity.

Comment (arno-lunarg) [reply layers/error_message/logging.cpp:380]: 🌴

**Comment by spencer-lunarg** [comment]
ok, no more spending time worrying about `emplace_back` vs `push_back` for at least a year now :upside_down_face:

**Pull request merged**